src/common/libidset/idset.h

```diff
@@ -0,0 +1,38 @@
+#ifndef _IDSET_H
+#define _IDSET_H
+
+#include <sys/types.h>
+
+/* An idset is an unordered set of non-negative integers  (0,1,2,3...).
```

Comment:
Is this really an unordered set or ordered set? 

If the input is "3,2,4,5", the output will become "3,2,4-5" or still "2-5"?

Reply:
should be 2-5.  Maybe "unordered" is the wrong way to describe it?  The order of insertion has no bearing on the order of the set - it is always encoded in sorted order.

Reply:
Thanks @garlick. 

Not a big deal, but I have a slight preference towards ordered set in that case.

In C++, 

[unordered_set](http://en.cppreference.com/w/cpp/container/unordered_set) doesn't keep sorted keys internally. So there isn't a good way to output sorted keys.

[set](http://www.cplusplus.com/reference/set/set/) is sorted and can output the keys sorted.